packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -706,6 +745,125 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
+                await tx.waitForDeployment();
+                return accountAlias.address; // Return the address for those that made deployments
+              }
+              return null;
+            });
+
+            const results = await Promise.all(deployPromises);
+            callingAccountAddresses.push(...results.filter((address) => address !== null));
+
+            expect(callingAccountAddresses.length).to.gt(0);
+
+            // awaiting for HBAR limiter to finish updating expenses in the background
+            await Utils.wait(6000);
+
+            // If an account has made a state-changing call, it will have a non-zero `amountSpent` in its associated spending plan.
+            // Otherwise, its `amountSpent` in the associated plan should remain 0.
+            for (const aliasAccountPlan of Object.values(accountPlanObject).flat()) {
+              const associatedSpendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(
+                aliasAccountPlan.hbarSpendingPlan.id,
+                requestDetails,
+              );
+
+              if (callingAccountAddresses.includes(aliasAccountPlan.aliasAccounts[0].address)) {
+                expect(associatedSpendingPlan.amountSpent).to.not.eq(0);
+              } else {
+                expect(associatedSpendingPlan.amountSpent).to.eq(0);
+              }
+            }
+          });
+
+          it(`Should eventually exhaust the total HBAR limits after many large contract deployments by different tiered users`, async () => {
+            const exchangeRateResult = (await mirrorNode.get(`/network/exchangerate`, requestId)).current_rate;
+            const exchangeRateInCents = exchangeRateResult.cent_equivalent / exchangeRateResult.hbar_equivalent;
+
+            const allAccountAliases = Object.values(accountPlanObject)
+              .flat()
+              .map((accountAliasPlan) => accountAliasPlan.aliasAccounts[0]);
+
+            let totalHbarSpent = 0;
+            const totalHbarBudget = ConfigService.get(`HBAR_RATE_LIMIT_TINYBAR`) as number;
+            const estimatedTxFee = estimateFileTransactionsFee(
+              largeContractJson.bytecode.length,
+              fileAppendChunkSize,
+              exchangeRateInCents,
+            );
+
+            while (totalHbarSpent + estimatedTxFee < totalHbarBudget) {
+              const promises = allAccountAliases.map(async (accountAlias) => {
+                try {
+                  for (let i = 0; i < 50; i++) {
+                    await deployContract(largeContractJson, accountAlias.wallet);
+                  }
+                  expect.fail(`Expected an error but nothing was thrown`);
+                } catch (e) {
+                  logger.error(e.message);
+                  expect(e.message).to.contain(predefined.HBAR_RATE_LIMIT_EXCEEDED.message);
+
+                  // awaiting for HBAR limiter to finish updating expenses in the background
+                  await Utils.wait(6000);
```

Comment:
Same here.

Reply:
In this case, since it's within a `while` loop and all transactions are initiated simultaneously using `Promise.all()`, it won’t wait for regular `async` but instead for `new Promise()`. Therefore, I need to keep the `wait` in this spot.